src/librustc/mir/mod.rs

```diff
@@ -2055,10 +2055,13 @@ impl<'tcx> Place<'tcx> {
 
     /// Finds the innermost `Local` from this `Place`.
     pub fn base_local(&self) -> Option<Local> {
-        match self {
-            Place::Base(PlaceBase::Local(local)) => Some(*local),
-            Place::Projection(box Projection { base, elem: _ }) => base.base_local(),
-            Place::Base(PlaceBase::Static(..)) => None,
+        let mut place = self;
+        loop {
+            match place {
+                Place::Projection(proj) => place = &proj.base,
+                Place::Base(PlaceBase::Static(_)) => return None,
+                Place::Base(PlaceBase::Local(local)) => return Some(*local),
+            }
```

Comment:
There's one awful way to write this more compactly:
```rust
pub fn base_local(mut self: &Self) -> Option<Local> {
    loop {
        match self {
            Place::Projection(proj) => self = &proj.base,
            Place::Base(PlaceBase::Static(_)) => return None,
            Place::Base(PlaceBase::Local(local)) => return Some(*local),
        }
    }
}
```

Reply:
I don't think it's really important. This code is going away at some point in the future where getting the base is just doing `place.base` or in this case `self.base`.